bin/forklift

```diff
@@ -0,0 +1,273 @@
+        installer = Installer(version['foreman'], version['katello'], version['puppet'],
+                              version['boxes'])
+
+        pipelines = deepcopy(default_pipelines)
+        if 'pipelines' in version:
+            merge(version['pipelines'], pipelines)
+
+        for product, pipeline in pipelines.items():
+            yield Segment(
+                product=product,
+                version=getattr(installer, product),
+                installer=installer,
+                common_roles=pipeline.get('common_roles', []),
+                bats_tests=pipeline.get('bats_tests', []),
+                hosts=pipeline.get('hosts', {}),
+            )
+
+
+def get_hosts(product, hosts, versions, box):
+    return {group: Host('pipeline-{}-{}-{}-{}'.format(product, group, versions, box), box, options['memory'])
+            for group, options in hosts.items()}
+
+
+def get_pipeline(config, product, versions, box):
+    segments = [s for s in get_segments(config) if s.product == product and s.version in versions]
+
+    if len(segments) != len(versions):
+        segment_versions = [segment.version for segment in segments]
+        missing = ', '.join(version for version in versions if version not in segment_versions)
+        raise ValueError('Version {} not found'.format(missing))
+
+    for segment in segments:
+        if box not in segment.installer.boxes:
+            raise ValueError("Box '{}' is not supported in {}".format(box, segment))
+
+    return Pipeline(
+            name='pipeline-{}-{}'.format(product, '-'.join(versions)),
+            filename='pipeline_{}_{}.yml'.format(product, '_'.join(versions).replace('.', '')),
+            product=product,
+            segments=sorted(segments),
+            hosts=get_hosts(product, segments[-1].hosts, '-'.join(versions), box),
+    )
+
+
+def play(hosts, variables, roles, become=True):
+    return {
+        'hosts': hosts,
+        'become': become,
+        'vars': variables,
+        'roles': roles,
+    }
+
+
+def run_playbook(path, *options, **variables):
+    var_options = (('-e', '{}={}'.format(key, value)) for key, value in variables.items())
+    command = ['ansible-playbook', path] + list(options) + list(*var_options)
+    return subprocess.check_call(command)
+
+
+def run_pipeline(path, destroy):
+    try:
+        run_playbook(path.as_posix(), forklift_state='up')
+    except subprocess.CalledProcessError:
+        if destroy == 'always':
+            click.echo('Playbook execution failed')
+        else:
+            raise click.ClickException('Playbook execution failed')
+
+    if destroy != 'never':
+        try:
+            run_playbook(path.as_posix(), forklift_state='destroy')
+        except subprocess.CalledProcessError:
+            raise click.ClickException('Shiva failed to destroy')
+
+
+@click.group()
+def forklift():
+    pass
+
+
+@forklift.command()
+@click.argument('product', default='katello', type=click.Choice(['katello', 'foreman']))
```

Comment:
At first glance, product seems like an odd choice of words since we refer to these as projects in all other places.